### PR TITLE
Add test to validate proposals

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description
- Add proposal helpers
- Add proposal tests

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7" target="_blank">Graphite</a> 
      * **PR #9** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/9" target="_blank">Graphite</a> 
        * **PR #11** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11" target="_blank">Graphite</a>  👈
          * **PR #12** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/12" target="_blank">Graphite</a> 
            * **PR #13** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/13" target="_blank">Graphite</a> 
              * **PR #14** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/14" target="_blank">Graphite</a> 
                * **PR #15** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/15" target="_blank">Graphite</a> 
                  * **PR #17** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/17" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11?utm_source=stack-comment).

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/7" target="_blank">Graphite</a> 
      * **PR #9** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/9" target="_blank">Graphite</a> 
        * **PR #11** <a href="https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/pooltogether-governor-upgrade/11?utm_source=stack-comment).

File: src/PooltogetherGovernor.sol [hunk 1]
```diff
@@ -70,6 +70,16 @@ contract PooltogetherGovernor is
   {
     return GovernorTimelockCompound.supportsInterface(interfaceId);
   }
+  /// @dev We override this for it to show up in our scopelint spec
+
+  function propose(
```
Comment (apbendi) [inline src/PooltogetherGovernor.sol:75]: Why did we need this now when we previously didn't 🤔

File: test/helpers/ProposalTest.sol [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import {Propose} from "script/Propose.s.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+import {PooltogetherGovernorTest} from "test/helpers/PooltogetherGovernorTest.sol";
+
+abstract contract AlphaProposalTest is PooltogetherGovernorTest {
+  //----------------- State and Setup ----------- //
+
+  IGovernorAlpha governorAlpha = IGovernorAlpha(GOVERNOR_ALPHA);
+  IERC20 usdcToken = IERC20(USDC_ADDRESS);
+  IERC20 radToken = IERC20(RAD_ADDRESS);
```
Comment (apbendi) [inline test/helpers/ProposalTest.sol]: We can remove this token (or both of them for now, until we get to testing them)

File: test/helpers/ProposalTest.sol [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+  function _assumeReceiver(address _receiver) internal {
+    assumePayable(_receiver);
+    vm.assume(
+      // We don't want the receiver to be the Timelock, as that would make our
+      // assertions less meaningful -- most of our tests want to confirm that
+      // proposals can cause tokens to be sent *from* the timelock to somewhere
+      // else.
+      _receiver != TIMELOCK
+      // We also can't have the receiver be the zero address because GTC
```
Comment (apbendi) [inline test/helpers/ProposalTest.sol]: We should see if this is the case for POOL and then update the comment or remove the `assume` statement

File: test/helpers/ProposalTest.sol [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import {
+  ICompoundTimelock
+} from "@openzeppelin/contracts/governance/extensions/GovernorTimelockCompound.sol";
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+
+import {Propose} from "script/Propose.s.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+import {PooltogetherGovernorTest} from "test/helpers/PooltogetherGovernorTest.sol";
+
+abstract contract AlphaProposalTest is PooltogetherGovernorTest {
```
Comment (apbendi) [inline test/helpers/ProposalTest.sol]: Since the contract is called `AlphaProposalTest`, should we name the file the same?

**Review by apbendi — COMMENTED** [review]
I know it's in draft and I haven't done a full review, just some initial quick comments!

File: src/PooltogetherGovernor.sol (hunk 1 above)
Comment (alexkeating) [reply src/PooltogetherGovernor.sol:75]: We don't need to do this, but the reason I did was to integrate the tests better with `scopelint spec`. It makes it easy to see exactly what test cases we cover. The output of the command in a future looks like the below.

Screenshot from 2023-07-11 17-53-18

File: test/helpers/ProposalTest.sol (hunk 4 above)
Comment (alexkeating) [reply test/helpers/ProposalTest.sol]: Good question, after moving a little past this I don't think we need the Alpha here. `ProposalTest` will do and I will update.

File: test/helpers/ProposalTest.sol (hunk 2 above)
Comment (alexkeating) [reply test/helpers/ProposalTest.sol]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/11/commits/33df525e4b15b3c0126dfa3caa34171e77cd36a8

File: test/helpers/ProposalTest.sol (hunk 3 above)
Comment (alexkeating) [reply test/helpers/ProposalTest.sol]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/11/commits/81f944ecd72ffe679caa0fcfc4f3e2d91e90bd7e

File: test/helpers/ProposalTest.sol (hunk 4 above)
Comment (alexkeating) [reply test/helpers/ProposalTest.sol]: https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/11/commits/c479e14f99c451b7b828025671756864e01e948e

**Comment by alexkeating** [comment]
Should be good after updating the env vars in the ci

File: test/helpers/ProposalTest.sol (hunk 3 above)
Comment (apbendi) [reply test/helpers/ProposalTest.sol]: I think we linked to the Governor contract when we meant to link to the POOL contract?

**Review by apbendi — APPROVED** [review]
Looking good, thanks @alexkeating

**Pull request merged**